service/route53/api_client.go

```diff
@@ -205,188 +205,188 @@ func addSanitizeURLMiddleware(stack *middleware.Stack) {
 // another operation (e.g. it expects just '1234')
 func sanitizeHostedZoneIDInput(input interface{}) error {
 	switch i := input.(type) {
-	case *ChangeResourceRecordSetsInput:
+	case *AssociateVPCWithHostedZoneInput:
 		if i.HostedZoneId != nil {
-			values := strings.SplitN(*i.HostedZoneId, "/", 2)
+			values := strings.Split(*i.HostedZoneId, "/")
```

Comment:
General feedback not specific to this PR. We should consider generating individual functions per operation input instead of a large switch statement. Having a large switch statement on types like these probably isn't super cheap, and static functions per operation wouldn't add significant source code.

Reply:
would take up a separate task to do this, as similar refactor would be needed in S3. 

Created https://github.com/aws/aws-sdk-go-v2/issues/847